middle_end/closure/closure.ml

```diff
@@ -882,8 +887,20 @@ let close_approx_var { fenv; cenv } id =
   match approx with
     Value_const c -> make_const c
   | approx ->
-      let subst = try V.Map.find id cenv with Not_found -> Uvar id in
-      (subst, approx)
+      match cenv with
+      | Not_in_closure -> Uvar id, approx
+      | In_closure { entries; env_param; env_pos } ->
+        let subst =
+          match V.Map.find id entries with
+          | Free_variable fv_pos ->
+            Uprim(P.Pfield (fv_pos - env_pos),
```

Comment:
This line and the following two lines needs a review since it differed from 5.2.0 and caused a merge conflict.